pandas/core/construction.py

```diff
@@ -807,6 +807,8 @@ def _try_cast(
         )
 
     elif dtype.kind in "mM":
+        if is_ndarray and arr.ndim == 2 and arr.shape[1] == 1:
```

Comment:
can you add a comment about why you are special-casing the arr.shape[1] == 1 case?